core/app/models/concerns/spree/default_cacheable.rb

```diff
@@ -0,0 +1,50 @@
+module Spree
+  module DefaultCacheable
+    extend ActiveSupport::Concern
+    included do
+      before_save :ensure_default
+      before_save :reload_cache
+    end
+
+    module ClassMethods
+      def default_cache_key
+        "#{Rails.application.class.parent_name.underscore}_default_#{name.demodulize.underscore}"
+      end
+
+      def default_query
+        column_names.include?('default') ?
+          where(default: true).first : column_names.include?('is_default') ?
```

Comment:
Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead.<br>Ternary operators must not be nested. Prefer `if`/`else` constructs instead.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.